### ISSUE TITLE
Hide scrollbars in WebView2

### ISSUE THREAD

**Issue by jpalo** [MicrosoftEdge/WebView2Feedback#188]
How would I hide scrollbars in WebView2, like with classic WebBrowser you set `ScrollBarsEnabled=false`?

WinForms, C#

**Comment by david-risney** [comment]
Great feature request! You might be able to inject CSS or script to hide scrollbars from within the web content, but we don't have a setting on the WebView2 API to turn off scrollbars.

Can you elaborate on your scenario? How are you using WebView2 and why is it useful to hide scrollbars in your case?

**Comment by dstaley** [comment]
In my particular case, I'd prefer to hide the scrollbars to prevent them from rendering. which disrupts the experience when the app is in dark mode since the scrollbars are white.

**Comment by jpalo** [comment]
Experience disruption here as well, picture speaks more than words. Any tips how to 
![image](https://user-images.githubusercontent.com/6917905/82416012-0cd60500-9a82-11ea-8b8f-ffa8d6f1835d.png)

This is how I injected CSS with previous WebBrowser control to modify some other things, I'd be happy to receive tips if similar think could be done with WebView2. In my case injecting JS would work as well.

```
private void SolarWebBrowser_DocumentCompleted(object sender, WebBrowserDocumentCompletedEventArgs e)
{
    IHTMLDocument2 doc = (solarWebBrowser.Document.DomDocument) as IHTMLDocument2;
    // The first parameter is the url, the second is the index of the added style sheet.
    IHTMLStyleSheet ss = doc.createStyleSheet("", 0);

    ss.cssText = Properties.Settings.Default.SolarGraphAdditionalStyles.Replace(";#GT;#", ">");
}
```

**Comment by david-risney** [comment]
We have CoreWebView2.ExecuteScriptAsync that will execute whatever JavaScript you pass in. Although I'm not sure what the equivalent of the code you have there would be in JS.

At any rate, we'll look into a setting to hide scrollbars.

Thanks!

**Comment by jpalo** [comment]
Of course, CoreWebView2.ExecuteScriptAsync worked great and was able to do all the things I needed including setting body overflow style to hidden to hide the scroll bars.

I will close this and leave it for you to decide if separate scrollbar hiding property is needed, Iäm leaning towards it not being needed.

**Comment by mikemeinz** [comment]
I see this is closed but I would like to add my request for the ability to disable or hide scrollbars. My Windows Forms app displays content on a wall-mounted TV Monitor using WebView2. No reason for scrollbars to occupy screen space in this scenario.

Thanks.

**Comment by champnic** [comment]
@mikemeinz The workaround works, but you'd like an easier API, is that correct?

**Comment by julesx** [comment]
an API for this would be great.

**Comment by 80lvlpalladin** [comment]
I would like to request an ability to hide scrollbars but retain scroll functionality. Default scrollbars don't fit well with my application UI.

**Comment by darbid** [comment]
I too would like to see the scroll bars removed. My use case is that I would like my WPF window not to be inflicted with an ugly 1980's styled scroll bar.

**Comment by champnic** [comment]
@darbid What's not to love about a gray rectangle? It probably saves battery life, or something :P 

I didn't realize this was still marked as closed, so reopening and it's not tracked on our backlog. Thanks!

**Comment by mikemeinz** [comment]
> @mikemeinz The workaround works, but you'd like an easier API, is that correct?

Yes!

**Comment by heering** [comment]
We also would highly appreciate the API option for hiding scrollbars as we are using the WebView2 control on a small touchscreen.

@jpalo How did you manage to hide scrollbars using CoreWebView2.ExecuteScriptAsync()? Unfortunately I'm not familiar with JavaScript ...

**Comment by jpalo** [comment]
@heering You need to use the NavigationCompleted event handler and execute a simple javascript. Details here: https://blog.jussipalo.com/2021/02/webview2-how-to-hide-scrollbars.html

**Comment by heering** [comment]
@jpalo Thank you very much. You saved my day 💯

**Comment by heering** [comment]
@jpalo I now recogized that, with this workaround, scrolling is disabled at all. I just want to hide the scrollbars because our users want to scroll by swiping the touchscreen, so I tried a different script: 

ExecuteScriptAsync("document.querySelector('body').setAttribute('style', '-ms-overflow-style: none; ');");

This style property is MS specific: https://reference.codeproject.com/Book/css/-ms-overflow-style

But without success. Do you have any idea? Is the syntax incorrect?

**Comment by champnic** [comment]
@heering The "-ms" prefix will only work with Internet Explorer and the old Microsoft Edge. The new Microsoft Edge is based on Chromium, so according to [this page](https://www.geeksforgeeks.org/hide-scroll-bar-but-while-still-being-able-to-scroll-using-css/) you can try the `-webkit-scrollbar` CSS psuedo-element.
https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar

**Comment by heering** [comment]
@champnic I tried this (any many other syntax alternatives) with no luck: 
wv.ExecuteScriptAsync("document.querySelector('body').style['-webkit-scrollbar'] = 'display: none'");
I'm not used to JavaScrpt, so maybe there is a syntax problem.

**Comment by champnic** [comment]
I'm not familiar with the syntax here myself, but StackOverflow may be able to help:
https://stackoverflow.com/questions/311052/setting-css-pseudo-class-rules-from-javascript

You can also right-click > Inspect on your page to get access to devtools and try out javascript in the Console tab. This should enable a faster test loop than recompiling with a different ExecuteScriptAsync script.

**Comment by jpalo** [comment]
The correct code to run is `document.querySelector('body').style.overflow='scroll';var style=document.createElement('style');style.type='text/css';style.innerHTML='::-webkit-scrollbar{display:none}';document.getElementsByTagName('body')[0].appendChild(style)` but for some reason injecting new STYLE nodes isn't working in `ExecuteScriptAsync`. You can test that by Inspecting the WebView2 and running the JS manually in Console.

**Comment by jpalo** [comment]
You mean it is working for you?

**Comment by ukandrewc** [comment]
Not sure it's doing it specifically for me ;-)

**Comment by heering** [comment]
This snippet is working for me, too! Thanks a lot :)

**Comment by microspaze** [comment]
Auto hide scrollbar for all web pages:

```
webView2.CoreWebView2InitializationCompleted += (sender, args) =>
            {
                if (args.IsSuccess && webView.CoreWebView2 != null)
                {
                    webView.CoreWebView2.DOMContentLoaded += (s, a) =>
                        {
                            //隐藏滚动条：https://blog.jussipalo.com/2021/02/webview2-how-to-hide-scrollbars.html
                            webView.ExecuteScriptAsync("document.querySelector('body').style.overflow='scroll';var style=document.createElement('style');style.type='text/css';style.innerHTML='::-webkit-scrollbar{display:none}';document.getElementsByTagName('body')[0].appendChild(style)");
                        };
                }
            };
```

**Comment by mikemeinz** [comment]
I removed scroll bars by including this in my style sheet:
     `body{overflow:hidden;text-overflow:clip;white-space: nowrap;}`

I'm no CSS expert so if anyone knows a better method, please comment.

**Comment by acrolink** [comment]
Seems no solution will work to disable the scrolling when viewing a PDF inside the webview2 browser.
